pkg/controller/loadtest.go

```diff
@@ -523,3 +510,17 @@ func checkLoadTestLifeTimeExceeded(loadTest *loadTestV1.LoadTest, deleteThreshol
 
 	return false
 }
+
+func (c *Controller) deleteLoadTest(ctx context.Context, key string, loadTest *loadTestV1.LoadTest) {
+	err := c.kangalClientSet.KangalV1().LoadTests().Delete(ctx, loadTest.Name, metaV1.DeleteOptions{})
+	if err != nil {
```

Comment:
reducing nesting for the main logic may slightly improve readability, so this condition may be changed to

```go
if err == nil {
  return
}

// here comes the main logic on the top level